addons/mail/static/src/models/chat_window_manager/chat_window_manager.js

```diff
@@ -99,11 +99,12 @@ function factory(dependencies) {
                 chatWindow = this.env.models['mail.chat_window'].create({
                     manager: [['link', this]],
                     threadViewer: [['create', {
+                        shouldLoadMessages: false,
```

Comment:
If possible instead of this flag, it would be best if hidden chat windows didn't have a thread viewer linked. So the logic would be in thread cache, depending on whether there is a thread viewer linked or not.

Reply:
That what I first intended to do but I was wondering if the link with the thread won't be lost if we don't store it directly ?

Reply:
Good concern, this change might imply storing thread directly on chat window instead of being a related.

Reply:
And then threadviewer becomes a compute that depends on thread and isVisible, something like that.